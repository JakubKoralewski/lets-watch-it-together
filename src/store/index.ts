import { configureStore } from '@reduxjs/toolkit'
// https://github.com/vercel/next.js/tree/canary/examples/with-redux-toolkit

import exampleReducer from './lib/exampleSlice'

export default configureStore({
	reducer: {
		counter: exampleReducer
	},
	devTools: true,
})