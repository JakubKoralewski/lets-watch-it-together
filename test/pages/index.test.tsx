import React from 'react'
import { render } from '../testUtils'
import Index from '../../src/pages/index'
import * as redux from 'react-redux'
import store from 'store'
import * as nextRouter from 'next/router'

// jest.mock('node-fetch')
// eslint-disable-next-line @typescript-eslint/no-unused-vars
// import fetch from 'node-fetch'
//
//

import { act } from '@testing-library/react'
;(nextRouter as any).useRouter = jest.fn()
;(nextRouter.useRouter as jest.Mock<any, any>).mockImplementation(
	() => ({
		route: '/',
		prefetch: jest.fn().mockImplementation(async () => {}),
	})
)

// const spy = jest.spyOn(redux, 'useSelector')
// spy.mockReturnValue({ username: 'test' })

function renderWithProvider(elem: JSX.Element, options: unknown) {
	return render(
		<redux.Provider store={store}>{elem}</redux.Provider>,
		options
	)
}

describe('Home page', () => {
	beforeEach(() => {
		global.fetch = jest.fn().mockImplementation(() =>
			Promise.resolve(({
				json: () => Promise.resolve({}),
			} as any) as Response)
		)
	})
	afterEach(() => {
		jest.restoreAllMocks()
	})
	it('matches snapshot', async () => {
		const providers = {"github":{"id":"github","name":"GitHub","type":"oauth","signinUrl":"http://localhost:3000/api/auth/signin/github","callbackUrl":"http://localhost:3000/api/auth/callback/github"}};
		await act(async () => {
			const { asFragment } =
				renderWithProvider(<Index providers={providers}/>, {})

			expect(asFragment()).toMatchSnapshot()
		})
	})

	// it('clicking button triggers alert', () => {
	// 	const { getByText } = renderWithProvider(<Index />, {})
	// 	window.alert = jest.fn()
	// 	fireEvent.click(getByText('Go to the about page'))
	// 	expect(window.alert).toHaveBeenCalledWith(
	// 		'With typescript and Jest'
	// 	)
	// })
})
