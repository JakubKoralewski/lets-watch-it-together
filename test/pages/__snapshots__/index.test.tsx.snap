// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      class="makeStyles-paper-1"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
          />
        </svg>
      </div>
      <h1
        class="MuiTypography-root MuiTypography-h5"
      >
        Sign in
      </h1>
      <div
        class="makeStyles-form-3"
      >
        <div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Sign In with GitHub
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-5"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      >
        Copyright Â© 
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          href="https://github.com/JakubKoralewski/lets-watch-it-together/blob/main/LICENSE"
        >
          Let's Watch It Together
        </a>
         2021.
      </p>
    </div>
  </main>
</DocumentFragment>
`;
